internal/page/page.go

```diff
@@ -33,12 +34,12 @@ type Links struct {
 	Next     string `json:"next"`
 }
 type Metadata struct {
-	Total int `json:"total"`
+	Total int `json:"total,omitempty"`
 }
 
 type Info struct {
```

Comment:
My suggestion would be renaming this to `Metaroot` instead of `Info`, that would include `NewMetaroot`.

Alternatively, I suggest to rename Metadata to just Data (keeping variable and JSON field as "metadata") and Info to Metadata.

Finally, my third suggestion is simply move links into metadata as this sets a bad practice - it feel like it is okay to pollute the root JSON namespace with fields while a single "metadata" field could be a nice solution with no other fields added.